ckanext/oidc_pkce/views.py

```
@@ -133,7 +133,9 @@ def callback():
         config.userinfo_url(),
         headers={"Authorization": f"Bearer {access_token}"},
     ).json()
-
+    if access_token:
```

If we're not sure whether an access token exists, shouldn't that affect the `Authorization` header in the previous request?

---

Hi @ThrawnCA  thanks for catching this!

You’re right, we shouldn’t send an Authorization header. I’ve updated the callback to fail fast when the token is absent (returning an error and redirect), and I only append the token to userinfo when it exists. The `/userinfo` request is now guaranteed to have a valid bearer token.

Thanks again for your review!